src/code/audio/audio_load.c

```diff
@@ -1,161 +1,2160 @@
+                    gAudioContext.sampleDmaReuseQueue2RdPos++;
+                }
+                dma->ttl = 32;
+                *dmaIndexRef = (u8)i;
+                return dma->ramAddr + (devAddr - dma->devAddr);
+            }
+        }
+
+        if (arg2 == 0) {
+            goto search_short_lived;
+        }
+
+        if (gAudioContext.sampleDmaReuseQueue2RdPos != gAudioContext.sampleDmaReuseQueue2WrPos && arg2 != 0) {
+            // Allocate a DMA from reuse queue 2, unless full.
+            dmaIndex = gAudioContext.sampleDmaReuseQueue2[gAudioContext.sampleDmaReuseQueue2RdPos];
+            gAudioContext.sampleDmaReuseQueue2RdPos++;
+            dma = gAudioContext.sampleDmas + dmaIndex;
+            hasDma = true;
+        }
+    } else {
+    search_short_lived:
+        dma = gAudioContext.sampleDmas + *dmaIndexRef;
+        i = 0;
+    again:
+        bufferPos = devAddr - dma->devAddr;
+        if (0 <= bufferPos && (u32)bufferPos <= dma->size - size) {
+            // We already have DMA for this memory range.
+            if (dma->ttl == 0) {
+                // Move the DMA out of the reuse queue, by swapping it with the
+                // read pos, and then incrementing the read pos.
+                if (dma->reuseIndex != gAudioContext.sampleDmaReuseQueue1RdPos) {
+                    gAudioContext.sampleDmaReuseQueue1[dma->reuseIndex] =
+                        gAudioContext.sampleDmaReuseQueue1[gAudioContext.sampleDmaReuseQueue1RdPos];
+                    gAudioContext
+                        .sampleDmas[gAudioContext.sampleDmaReuseQueue1[gAudioContext.sampleDmaReuseQueue1RdPos]]
+                        .reuseIndex = dma->reuseIndex;
+                }
+                gAudioContext.sampleDmaReuseQueue1RdPos++;
+            }
+            dma->ttl = 2;
+            return dma->ramAddr + (devAddr - dma->devAddr);
+        }
+        dma = gAudioContext.sampleDmas + i++;
+        if (i <= gAudioContext.sampleDmaListSize1) {
+            goto again;
+        }
+    }
+
+    if (!hasDma) {
+        if (gAudioContext.sampleDmaReuseQueue1RdPos == gAudioContext.sampleDmaReuseQueue1WrPos) {
+            return NULL;
+        }
+        // Allocate a DMA from reuse queue 1.
+        dmaIndex = gAudioContext.sampleDmaReuseQueue1[gAudioContext.sampleDmaReuseQueue1RdPos++];
+        dma = gAudioContext.sampleDmas + dmaIndex;
+        hasDma = true;
+    }
+
+    transfer = dma->size;
+    dmaDevAddr = devAddr & ~0xF;
+    dma->ttl = 3;
+    dma->devAddr = dmaDevAddr;
+    dma->sizeUnused = transfer;
+    AudioLoad_Dma(&gAudioContext.currAudioFrameDmaIoMesgBuf[gAudioContext.curAudioFrameDmaCount++], OS_MESG_PRI_NORMAL,
+                  OS_READ, dmaDevAddr, dma->ramAddr, transfer, &gAudioContext.currAudioFrameDmaQueue, medium,
+                  "SUPERDMA");
+    *dmaIndexRef = dmaIndex;
+    return (devAddr - dmaDevAddr) + dma->ramAddr;
+}
+
+const char D_801E030C[] = "TYPE %d:ID %d is not External Map.\n";
```

Comment:
Is this just an unused string literal that somehow found its way into rodata? If so the current way we have been handling it is using is as `(void)"string literal"` where we feel it makes the most sense. For an example see `__osCheckArena`.

Reply:
As was brought up in discord, we looked for a possible place for this string to go, but it doesn't seem to have any place to go that makes sense. I can add the (void) stuff to it, but any thoughts as to where this should go? Or should I just leave it outside like this?

Reply:
So the `(void)` trick doesn't seem to work unless this is inside a function. And since the contents of the string don't match up with any function, I'm going to leave this as is unless there's another suggestion as to how to handle this string

Reply:
Yeah that sounds fine. Could you at least add a note that it doesn't seem to fit anywhere?